platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/jos/JavaObjectSerializationCodec.kt

```diff
@@ -82,6 +83,12 @@ class JavaObjectSerializationCodec(
     override suspend fun ReadContext.decode(): Any =
         decodePreservingIdentity { id ->
             when (readEnum<Format>()) {
+                Format.Externalizable -> {
```

Comment:
❓ Did you consider making it a top-level encoder instead?

Reply:
I did consider that. I figured `Externalizable` is closely related to the rest of the Java Object Serialization (and also a `Serializable`, so there would be benefit in keeping it next to the support for other kinds of `Serializable` types.

Reply:
I see. I'm ambivalent about the runtime cost of the setup, namely, the double `instanceof` check on every `Serializable` instance plus, and more importantly, the BuildTree scoped cache entry in `JavaSerializationEncodingLookup`. I'm leaning slightly towards a top-level codec.

Reply:
There's also a data cost to the unnecessary indirection.